tests/test_demography.py

```diff
@@ -3060,6 +3060,38 @@ def test_old_style_metadata_via_sim_ancestry(self):
         }
         assert expected == pop.metadata
 
+    def test_old_style_metadata_description_is_merged(self):
+        md = {"description": "y"}
+        demography = msprime.Demography.from_old_style(
+            population_configurations=[
+                msprime.PopulationConfiguration(initial_size=1, metadata=md)
+            ]
+        )
+        ts = msprime.sim_ancestry(2, demography=demography, random_seed=1)
+        assert ts.num_populations == 1
+        pop = ts.population(0)
+        expected = {
+            "name": "pop_0",
+            "description": "y",
+        }
+        assert expected == pop.metadata
+
+    def test_old_style_metadata_name_is_merged(self):
```

Comment:
Same test twice?

Reply:
One test for the `name`, one for the `description` (the only two "standard" metadata fields). If the `name` field is in the metadata, this hits an additional code path in Demography.from_old_style(). But I can merge the two functions into one if you'd prefer that?

Reply:
Ah yes, my bad. Ta.